src/libraries/System.Collections/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -55,5 +56,15 @@ public static class CollectionExtensions
             value = default;
             return false;
         }
+
+        public static ReadOnlyCollection<T> AsReadOnly<T>(this IList<T> list)
+        {
+            return new ReadOnlyCollection<T>(list);
+        }
+
+        public static ReadOnlyDictionary<TKey, TValue> AsReadOnly<TKey, TValue>(this IDictionary<TKey, TValue> dictionary)
```

Comment:
Add where TKey:not null

Reply:
This is not a requirement for IDictionary<TKey, TValue> or the read-only class.

Reply:
Actually, I'm wrong. It is a requirement for `ReadOnlyDictionary<TKey, TValue>`

Reply:
fixed